### DISCUSSION TITLE
Phone & Email SignUp

### DISCUSSION THREAD

**Discussion by Anan7Codes** [supabase/supabase#5631] [Questions] [answered]
Hey guys, I am trying to build a signup flow where the user signups entering his/her email, password & phone number. They've to obv confirm both email and phone number as well but be able to sign in only using their email. How can I achieve this?

**Comment by ltanady** [comment, marked as answer]
If you want to sign up the user with both email and phone, you'll need to sign up with either email + password or phone + password then update the other missing field via the **PUT /users** endpoint. 

For now, it does not trigger the OTP verification when you update the missing field via **PUT /users** endpoint. We are still working on making that happen, you can susbcribe to the issue here:
https://github.com/supabase/gotrue/issues/390

**Reply by Anan7Codes** [reply]
Alright Thanks!!

**Comment by ColeTownsend** [comment]
Looks like that issue was closed https://github.com/supabase/gotrue/pull/421, so is there a good way to do this now? If you allowed sign up with phone + OTP, the only way to add email and password is still via a rest call

**Comment by SeonaidhMorrison** [comment]
Hi there, still not sure how i can use both email and password to authenticate users on my flutter app. Logs give me the error ' "email" or "phone" needs to be specified'
